configure.ac

```diff
@@ -540,6 +540,13 @@ fi
 # Checks for supported compiler flags
 AX_APPEND_COMPILE_FLAGS([-Wall -Wformat -Wformat-security])
 
+# Place the functions and data into separate sections, allowing the linker
+# to garbage collect the unused ones.
+AX_APPEND_COMPILE_FLAGS([-ffunction-sections -fdata-sections])
+
+GC_SECTIONS="-Wl,--gc-sections"
+AC_SUBST(GC_SECTIONS)
```

Comment:
It would be nice to check that the system linker supports these flags before unconditionally including them; is there an easy way to do that?

Reply:
From memory they should be supported with gcc4 onward. Will add some test though - especially since the MacOS builds fail

Reply:
The autotools part is trivial, the cmake one less-so...

We require cmake 2.8.12, for which there are no official docs on cmake.org. The closest one 3.0 shows that the standard `check_linker_flag` API is missing and even the old version of `try_compile()` (that we currently use) lacks support for linker flags. Guess I'll hack it for the time being :grinning:

Reply:
Added linker flag check for autotools, used "CMAKE_SYSTEM_NAME != Darwin" - since there's no obvious way to check.